core/templates/components/button-directives/primary-button.component.html

```diff
@@ -1,5 +1,14 @@
-<button type="submit"
-        class="oppia-primary-button"
+<a *ngIf="!componentIsButton" class="primary-button btn"
```

Comment:
The class name still matches the component name (cf this comment https://github.com/oppia/oppia/pull/19676#discussion_r1479535852). I suggest that perhaps you could wrap all this in a span and put a class on that span called "primary-button-container", or (perhaps better) drop the primary-button class altogether since you could just use the tag name as the selector in CSS.

Reply:
A class is indeed necessary for specificity and to not interfere with other styles on pages where the component is used. The class is called `primary-button` whereas the component name is `oppia-primary-button`, am I missing out on something?

Reply:
Sorry, you're not -- it was my error. I got thrown off because the file name here is primary-button.component.html, but I just realized we refer to it as oppia-primary-button in the tag when calling it. Should the name of the file be changed to match the name of the directive?

Re specificity: what is a case where you need to use `.primary-button` rather than `oppia-primary-button` as a CSS selector?

Reply:
No worries! I don't think the file names should be changed as they simply follow the order that all other components already have - like file name being `primary-button.component` and the selector being `oppia-primary-button`. 

Regarding specificity, since all have the `.btn` class by default, the styles of .btn can override our component's styles as `oppia-primary-button` is less specific and simply not enough.
For example, the color among other styles for this button on a page being overridden -
![image](https://github.com/oppia/oppia/assets/73329256/a40e8383-451c-4736-a549-aa04e67e5295)
![image](https://github.com/oppia/oppia/assets/73329256/189d3fb1-a54e-48f7-b0fa-fb84719b3cc7)

Reply:
@Vir-8 Ah, re specificity, got it. I suggest that you add a comment in the code to explain that that class is necessary for that reason.

And yeah, the filename thing is fine, I checked a random component and it actually does align with that convention. I'm not sure how I've missed that for so long -- thanks for pointing it out!

Reply:
Done, added a comment!